tests/services/githubevent.service.spec.ts

```diff
@@ -66,18 +66,21 @@ describe('GithubEventService', () => {
     it('clears the details of the most recent event', async () => {
       githubService.fetchEventsForRepo.and.returnValue(of(EVENTS));
       const githubEventService: GithubEventService = new GithubEventService(githubService, null);
+      const githubEventServiceSetLastModifiedTime = spyOn<any>(githubEventService, 'setLastModifiedTime');
+      const githubEventServiceSetLastModifiedCommentTime = spyOn<any>(githubEventService, 'setLastModifiedCommentTime');
+
       await githubEventService.setLatestChangeEvent().toPromise();
-      expect(githubEventService.getLastModifiedTime()).toBeDefined();
-      expect(githubEventService.getLastModifiedCommentTime()).toBeDefined();
+      expect(githubEventServiceSetLastModifiedTime).toHaveBeenCalledTimes(1);
+      expect(githubEventServiceSetLastModifiedCommentTime).toHaveBeenCalledTimes(1);
       githubEventService.reset();
-      expect(githubEventService.getLastModifiedTime()).toBeUndefined();
-      expect(githubEventService.getLastModifiedCommentTime()).toBeUndefined();
+      expect(githubEventServiceSetLastModifiedTime).toHaveBeenCalledWith(undefined);
+      expect(githubEventServiceSetLastModifiedCommentTime).toHaveBeenCalledWith(undefined);
     });
   });
 
 });
 
 function assertLastModified(githubEventService: GithubEventService, expectedEvents) {
-  expect(githubEventService.getLastModifiedTime()).toBe(expectedEvents[0].created_at);
-  expect(githubEventService.getLastModifiedCommentTime()).toBe(expectedEvents[0].issue.updated_at);
+  expect(githubEventService['lastModified']).toBe(expectedEvents[0].created_at);
```

Comment:
I am not sure if the refactor has a strong benefit, if we have to access the property directly.

However, you are right that (ideally) we should not inspect the private state of the service in the test.
I think we can consider tweaking the public API of GithubEventServich such that the `reload` method returns `Observable<boolean>` (true if re-loading of issues via IssueService was triggered, false otherwise).

In our tests, we can then verify whether `setLatestChangeEvent`, `reload` and `reset` methods work correctly by observing the returned status from `reload`.   We will no longer need to use the getters, in this way.

Reply:
I see then we wouldn't need to return the json data of the latest event i assume based on the documentation of `reloadPage()` and instead now we would return a `Observable<boolean>` instead. 

I have a question. assuming `reloadAllIssues()` are called, it would make sense to map it into an Observable to true like this? 

`return this.issueService.reloadAllIssues().pipe(flatMap((response: any[]) => of(true)));`

Else, we would put the other control statements to return `of(false)` since no changes were made from the last reload.

Reply:
Yes, we can return `Observable<boolean>` and the mapping is good as well

Reply:
👍